ddtrace/debugging/_exception/replay.py

```diff
@@ -188,7 +190,9 @@ def on_span_exception(
                         )
 
                         # Capture
-                        snapshot.line()
+                        snapshot.line(ChainMap(frame.f_locals, frame.f_globals))
+
+                        snapshot.state = SignalState.DONE
 
                         # Collect
                         self.__uploader__.get_collector().push(snapshot)
```

Comment:
where does the rate-limit of exception replay happen?

Reply:
rate-limiting is done at the very beginning https://github.com/DataDog/dd-trace-py/blob/1d33870b84922eaf89136b75e7e90a048f3fa688/ddtrace/debugging/_exception/replay.py#L156-L157